### PR TITLE
Ensure unique addresses for GATEWAY_LISTEN

### PR DESCRIPTION
**Pull request by simonrw**
## Motivation

If `LOCALSTACK_HOST` is set to include port 443, and `GATEWAY_LISTEN` is not set, in -ext we by default add port 443 to the list, so `GATEWAY_LISTEN` becomes:

```
GATEWAY_HOST = [
    HostAndPort("0.0.0.0", 443),
    HostAndPort("0.0.0.0", 443),
]
```

Hypercorn then tries to bind to the port twice and crashes.

## Changes

* Introduce the concept of a `UniqueHostAndPortList` - see the docstring for guarantees
* Configure the dev container to modify the edge port config

### CODE REVIEW

File: localstack/dev/run/__main__.py [hunk 1]
```diff
@@ -234,6 +234,15 @@ def run(
...
 
+    # replicate pro startup
+    if pro:
+        try:
+            from localstack_ext.plugins import modify_edge_port_config
+
+            modify_edge_port_config(config)
+        except ImportError:
+            pass
+
```
Comment (dfangl) [inline localstack/dev/run/__main__.py:245]: Why is this necessary? Shouldn't this take place in the container then anyway?

**Review by dfangl — APPROVED** [review]
Although I do not like that a "wrong" configuration like `GATEWAY_LISTEN=:4566,:4566` works now, it seems more user friendly this way :)

File: localstack/dev/run/__main__.py (hunk 1 above)
Comment (simonrw) [reply localstack/dev/run/__main__.py:245]: It is in a `configure_localstack_container` hook which is not run in this dev container configuration cc @thrau

Comment (dfangl) [reply localstack/dev/run/__main__.py:245]: Ah, you are right, it is usually called as configure hook as well. thanks for the clarification!
